### PR TITLE
[DDIDO-538]: Add nginx redirect support.

### PR DESCRIPTION
**Pull request by steveworley**
- Adds an onbuild directive for copying in a sites redirect configuration

### CODE REVIEW

**Comment by steveworley** [comment]
There is a dependency on this with https://github.com/dpc-sdp/sdp-ansible-upgrade/pull/60 - the redirect-maps.conf file needs to be in project repositories before the `ONBUILD` directive is merged.

File: docker-compose.yml [hunk 1]
```diff
@@ -33,6 +33,22 @@ services:
     networks:
       - default
 
+  nginx_goss:
```
Comment (anthony-malkoun) [inline docker-compose.yml:52]: Goss is for testing images yeah? Should there be a test as well or is this for future testing?

Comment (steveworley) [reply docker-compose.yml:52]: Yeah goss is for image testing - we can include a more substantive test for this particular feature though (something like the clamav tests).

In this container we mount a redirect file, make a HTTP request and verify the response code. Will update the PR.

Comment (steveworley) [reply docker-compose.yml:52]: The idea of the nginx_goss containers is to make sure that during CI the `ONBUILD` step works.

**Comment by nicksantamaria** [comment]
https://github.com/dpc-sdp/sdp-ansible-upgrade/pull/60 is merged - waiting for that to get rolled out to all the projects before merging this. Then will need to retrofit the content-health migrate map to this one.

**Review by nicksantamaria — APPROVED** [review]

**Comment by nicksantamaria** [comment]
I did some more digging to figure out whether this PR was ok to merge. The key issue with this PR, is if the `redirects-map.conf` file is missing in the drupal project, the docker build will fail. So we need to distribute this file to all projects prior to adding this change to the base image.

Currently only 5 projects have the file. I tried to figure out why, and it turns out the 1.37.0 release was run on most projects _a month prior_ to the file being added to the sdp-platform ansible role - https://github.com/dpc-sdp/ansible-role-sdp-tide-platform/commit/aa59c3f3381ec7e6a195865518655e3d1466367e

So looks like we need to wait for 1.38.0 to roll out on all projects before this can proceed. Either that, or create a `5.x` branch as this is technically a backward compatibility break.

Alternatively, we could run the `ADD` statement in the project dockerfiles.

**Pull request closed without merging**